packaging/linux/common/khiops_env.in

```diff
@@ -79,16 +82,29 @@ if [ -f "/usr/share/khiops/khiops.jar" ]; then
     fi
 fi
 
-@ADDITIONAL_EN_VAR@
+get_script_dir(){
+    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+    echo "$SCRIPT_DIR/"
+}
+@ADDITIONAL_ENV_VAR@
+
+# Setting up mpi for multiple machines
+if [[ -n $KHIOPS_MPI_HOST_FILE ]]; then
+    MPI_EXTRA_FLAGS="@MPIEXEC_HOSTFILE_FLAG@ $KHIOPS_MPI_HOST_FILE"
+    @EXPORT_ENV_VARIABLES@
+fi
+
+KHIOPS_PATH=@KHIOPS_PATH@@MODL_NAME@
+KHIOPS_COCLUSTERING_PATH=@KHIOPS_COCLUSTERING_PATH@MODL_Coclustering
 
 # Number of processes in use (must be set according to the physical cores number)
 if [[ -z $KHIOPS_PROC_NUMBER ]]; then
-    KHIOPS_PROC_NUMBER=$(lscpu -b -p=Core,Socket | grep -v '^#' | sort -u | wc -l)
+    KHIOPS_PROC_NUMBER=$("@GET_PROC_NUMBER_PATH@"_khiopsgetprocnumber | head -n 1 )
 fi
 
 if command -v mpiexec &> /dev/null
 then
-    KHIOPS_MPI_COMMAND="mpiexec @KHIOPS_MPI_EXTRA_FLAG@ @MPIEXEC_NUMPROC_FLAG@ $KHIOPS_PROC_NUMBER"
+    KHIOPS_MPI_COMMAND="mpiexec $MPI_EXTRA_FLAGS @KHIOPS_MPI_EXTRA_FLAG@ @MPIEXEC_NUMPROC_FLAG@ $KHIOPS_PROC_NUMBER"
```

Comment:
IMHO, at least on Conda, we should provide the full path to `mpiexec`, rather than relying on `which` for finding the good `mpiexec` (at least on Conda). Because if e.g. MPI somehow got uninstalled (or corrupted) in the Conda environment, I'm not sure we want to silently fall back on the system-wide MPI; we'd rather issue warning to the user that MPI is not available. Especially that the MPI might not be the same between Conda (mpich) and native (OpenMPI).